server/conn.go

```diff
@@ -1153,23 +1153,7 @@ func (cc *clientConn) handleLoadData(ctx context.Context, loadDataInfo *executor
 	go processStream(ctx, cc, loadDataInfo)
 	err = loadDataInfo.CommitWork(ctx)
 	loadDataInfo.SetMessage()
-
-	var txn kv.Transaction
-	var err1 error
-	txn, err1 = loadDataInfo.Ctx.Txn(true)
-	if err1 == nil {
-		if txn != nil && txn.Valid() {
-			if err != nil {
-				if err1 := txn.Rollback(); err1 != nil {
-					logutil.Logger(ctx).Error("load data rollback failed", zap.Error(err1))
-				}
-				return err
-			}
-			return cc.ctx.CommitTxn(sessionctx.SetCommitCtx(ctx, loadDataInfo.Ctx))
-		}
-	}
```

Comment:
remove this snippet will remove function added by https://github.com/pingcap/tidb/pull/9444?

maybe some reason we need `txn.Rollback` here if `txn.Valid` and have `err`, PTAL @crazycs520

Reply:
former implementation will leave one batch/txn undecided out for loop, so the main logic need to  process the last txn

Reply:
PTAL @crazycs520

Reply:
@jackysp  PTAL